api-schema/src/main/java/org/stellar/anchor/api/callback/PutCustomerRequest.java

```diff
@@ -119,4 +123,9 @@ public class PutCustomerRequest {
   String ip_address;
 
   String sex;
+
+  public static PutCustomerRequest from(Sep12PutCustomerRequest request) {
+    Gson gson = new Gson();
+    return new Gson().fromJson(gson.toJson(request), PutCustomerRequest.class);
```

Comment:
Can we not rely on Gson for this? It's not clear to see how null fields, missing fields are treated without looking how the `Gson` object was initialized. If the default behavior changes between `Gson` versions, it might end up breaking the build.

Same comment for `to`/`from` methods in the other `*Request` classes.

Reply:
Nit: also this mapping seems like an implementation detail of `Sep12Service`, does it make sense to put this method there instead?

Reply:
If we're doing this elsewhere, then I won't block the PR on this. But it would be nice to not depend on `Gson` for these types of conversion.

Reply:
We use GSON for AP. We have a `GsonUtils` class that handles the conversion. 

This PR is focusing on fixing the security leak. I think we can add another refactor ticket tech-debt ticket to clean up the conversion pattern. I would not make this blocking the PR.